src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -1278,4 +1287,46 @@ public SFClientConfig getSfClientConfig() {
   public void setSfClientConfig(SFClientConfig sfClientConfig) {
     this.sfClientConfig = sfClientConfig;
   }
+
+  private void runDiagnosticsIfEnabled() throws SnowflakeSQLException {
+    /*
+    If the user enables diagnostics then we need to be sure we don't go on
+    and create an actual connection because that would be a waste of time since
+    they're trying to debug a connection issue. We need to throw a SqlException
+    back to the calling application with a clear message explaining that a connection
+    was not created.
+     */
+
+    Map<SFSessionProperty, Object> connectionPropertiesMap = getConnectionPropertiesMap();
+    boolean isDiagnosticsEnabled =
+        (connectionPropertiesMap.get(SFSessionProperty.ENABLE_DIAGNOSTICS) == null)
```

Comment:
Maybe we can replace it with `Optional`?

Reply:
Done